src/test/scala/mesosphere/mesos/TaskBuilderTest.scala

```diff
@@ -245,6 +245,180 @@ class TaskBuilderTest extends MarathonSpec with Matchers {
     assert(portsResource.getRole == "marathon")
   }
 
+  test("build creates task for DOCKER container using named, external [DockerVolume] volumes") {
+    val offer = MarathonTestHelper.makeBasicOffer(
+      cpus = 2.0, mem = 128.0, disk = 2000.0, beginPort = 31000, endPort = 32000
+    ).build
+
+    val task: Option[(MesosProtos.TaskInfo, Seq[Int])] = buildIfMatches(
+      offer,
+      AppDefinition(
+        id = "/product/frontend".toPath,
+        cmd = Some("foo"),
+        cpus = 1.0,
+        mem = 32.0,
+        executor = "//cmd",
+        portDefinitions = Nil,
+        container = Some(Container(
+          docker = Some(Docker()), // must have this to force docker container serialization
+          `type` = MesosProtos.ContainerInfo.Type.DOCKER,
+          volumes = Seq[Volume](
+            // TODO(jdef) probably need to validate uniquess of hostPaths's that reference volume
```

Comment:
discussed this with @sttts and identified possible workarounds. they all suck. the best thing we can do is to deprecate the use of volume names with hostPath and tell people to use external volumes if that's what they want.